### PR TITLE
Add new Documentation section to the Playbook

### LINKED ISSUE #533: Create a new Documentation section in the Playbook
Create a new Documentation section in the Playbook.

Table of Contents:
- Goals
- Challenges
- What documentation should exist
- Best practices
- Tools
- Resources

### PR DESCRIPTION
**Pull request by magencio**
## What are you trying to address

Add a new Documentation section in the Playbook. Closes #533.

## Description of new changes

Adds the main readme for the new section and initial contents for:

- Goals
- Challenges
- What documentation should exist
- Best practices
- Tools
- Recipes
- Resources

Most of these initial contents have been gathered from existing contents in the Playbook.

[Several backlog items have been created](https://github.com/microsoft/code-with-engineering-playbook/projects/1?card_filter_query=label%3Adocumentation) for new proposed content after this first version of the section gets merged. The idea of this first PR is to provide basic content and the skeleton for future content.

### CODE REVIEW

File: documentation/guidance/project-and-repositories.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - [Architecture Decision Record (ADRs)](../../design-reviews/decision-log/readme.md#architecture-decision-record-ADR)
+  - [Trade Studies](../../design-reviews/trade-studies/readme.md)
+
+Some sections in the documentation of the repository might point to the project’s documentation (e.g., Onboarding, Working Agreement, Contributing Guide).
+
+## Common documentation to all repositories
+
+- Introduction
+  - Project
+  - Customer
```
Comment (TessFerrandez) [inline documentation/guidance/project-and-repositories.md:28]:
```suggestion
  - Stakeholders
```

**Review by TessFerrandez — APPROVED** [review]
Looks great - just had some language nits

File: documentation/readme.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  - Documents not updated along with the code, so they don't mention the right folders, settings, etc.
+- **Obsolete**.
+  - Design documents that don't apply anymore sitting next to valid documents. Which one shows the latest decisions?
+- **Out of order (subject / date)**.
+  - Documents not organized per subject/workstream so not easy to find relevant information when you change to a new workstream.
+  - Design decision logs out of order and without a date that helps to determine which is the final decision on something.
+- **Duplicate**.
+  - No settings file available in a centralized place as a single source of truth, so developers must keep sharing their own versions and we end up with many different files that might or might not work.
+- **Afterthought**.
+  - Key documents created when we've been several weeks into the project: onboarding, how to run the app, etc.
```
[the remark this answers is not in the document]
Comment (magencio) [reply documentation/readme.md]: Changed

File: documentation/readme.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  - Documents that we create in the last minute just before the end of a project, forgetting that they also help the team while working on the project.
```
[the remark this answers is not in the document]
Comment (magencio) [reply documentation/readme.md]: Changed

File: documentation/best-practices/automation.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Replacing documentation with automation
+
+You could document e.g. how to setup your dev machine with the right version of the framework required to run the code, which extensions are useful to develop the application with your editor, or how to configure your editor to launch and debug the application. Or you could provide the means to automate tool installs, application startup, etc., instead.
```
[the remark this answers is not in the document]
Comment (magencio) [reply documentation/best-practices/automation.md]: Changed

File: documentation/guidance/project-and-repositories.md (hunk 1 above)
Comment (magencio) [reply documentation/guidance/project-and-repositories.md]: Changed

File: documentation/tools/automation.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# How to automate simple checks
+
```
Comment (bsherwin) [inline documentation/tools/automation.md:2]: Not sure if it is best to include here...but the team needs to be clear what "linting rules" are required and shouldn't be "overridden" with tooling or comments. Case in point the "markdown-link-check" that is currently blocking this PR...is that a "comment workaround" or fixing the link? The team should have consensus on when to override tooling rules.

**Review by bsherwin — APPROVED** [review]
I feel like some of this viewed as required is what gets our teams into tool version/install hell which is why it doesn't end up getting done. We should be clear as this fleshes out what is the MVP of documentation.

File: documentation/best-practices/establish-and-manage.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Establishing and managing documentation
+
+Documentation should be source controlled. [Pull Requests](../guidance/pull-requests.md) could then be used to tell others about the changes, so they can be reviewed and discussed. E.g., [Async Design Reviews](../../design-reviews/recipes/async-design-reviews.md).
```
[the remark this answers is not in the document]
Comment (magencio) [reply documentation/best-practices/establish-and-manage.md]: changed

File: documentation/tools/automation.md (hunk 5 above)
Comment (magencio) [reply documentation/tools/automation.md:2]: Thank you, I added your comments to the document, as I think they are relevant to the content

**Pull request merged**